receiver/signalfxreceiver/signalfxv2_to_metricdata.go

```diff
@@ -17,214 +17,190 @@ package signalfxreceiver
-			Points:      []*metricspb.Point{point},
-		}
-		metric := &metricspb.Metric{
-			MetricDescriptor: descriptor,
-			Timeseries:       []*metricspb.TimeSeries{ts},
-		}
-		metrics = append(metrics, metric)
+		i++
 	}
 
-	md.Metrics = metrics
-	return md, numDroppedTimeSeries
+	metrics.Resize(i)
+
+	return md, numDroppedDataPoints
 }
 
-func convertType(
+func fillInType(
 	sfxDataPoint *sfxpb.DataPoint,
-) (descType metricspb.MetricDescriptor_Type, err error) {
-
+	m pdata.Metric,
+) (err error) {
 	// Combine metric type with the actual data point type
 	sfxMetricType := sfxDataPoint.GetMetricType()
 	sfxDatum := sfxDataPoint.Value
 	if sfxDatum.IntValue == nil && sfxDatum.DoubleValue == nil && sfxDatum.StrValue == nil {
-		return metricspb.MetricDescriptor_UNSPECIFIED, errSFxNilDatum
+		return errSFxNilDatum
 	}
 
 	switch sfxMetricType {
 	case sfxpb.MetricType_GAUGE:
-		// Numerical: Periodic, instantaneous measurement of some state.
-		descType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+		if sfxDatum.DoubleValue != nil {
+			// Numerical: Periodic, instantaneous measurement of some state.
+			m.SetDataType(pdata.MetricDataTypeDoubleGauge)
+			m.DoubleGauge().InitEmpty()
+		}
 		if sfxDatum.IntValue != nil {
-			descType = metricspb.MetricDescriptor_GAUGE_INT64
+			m.SetDataType(pdata.MetricDataTypeIntGauge)
+			m.IntGauge().InitEmpty()
 		}
 
-	case sfxpb.MetricType_COUNTER, sfxpb.MetricType_CUMULATIVE_COUNTER:
-		// COUNTER:  Count of occurrences. Generally non-negative integers.
-		// CUMULATIVE_COUNTER: Tracks a value that increases over time, where
-		// only the difference is important.
-		descType = metricspb.MetricDescriptor_CUMULATIVE_DOUBLE
-		if sfxDatum.IntValue != nil {
-			descType = metricspb.MetricDescriptor_CUMULATIVE_INT64
+	case sfxpb.MetricType_COUNTER:
+		if sfxDatum.DoubleValue != nil {
+			m.SetDataType(pdata.MetricDataTypeDoubleSum)
+			m.DoubleSum().InitEmpty()
+			m.DoubleSum().SetAggregationTemporality(pdata.AggregationTemporalityDelta)
+		} else if sfxDatum.IntValue != nil {
+			m.SetDataType(pdata.MetricDataTypeIntSum)
+			m.IntSum().InitEmpty()
+			m.IntSum().SetAggregationTemporality(pdata.AggregationTemporalityDelta)
+		}
+
+	case sfxpb.MetricType_CUMULATIVE_COUNTER:
+		if sfxDatum.DoubleValue != nil {
+			m.SetDataType(pdata.MetricDataTypeDoubleSum)
+			m.DoubleSum().InitEmpty()
+			m.DoubleSum().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
+		} else if sfxDatum.IntValue != nil {
+			m.SetDataType(pdata.MetricDataTypeIntSum)
+			m.IntSum().InitEmpty()
+			m.IntSum().SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
 		}
 
 	case sfxpb.MetricType_ENUM:
 		// String: Used for non-continuous quantities (that is, measurements where there is a fixed
 		// set of meaningful values). This is essentially a special case of gauge.
 		// Attempt to treat it as a numeric gauge.
-		descType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+		m.SetDataType(pdata.MetricDataTypeDoubleSum)
```

Comment:
As the comment says, this should be IntGauge probably.

Reply:
Got rid of enum handling since it won't be seen.